LiteCore/Database/TreeDocument.cc

```diff
@@ -640,14 +644,51 @@ namespace litecore {
             return true;
         }
 
+        static bool hasEncryptables(slice body, SharedKeys* sk) {
+#ifndef COUCHBASE_ENTERPRISE
+            return false;
+#else
+            const Value* v = Value::fromTrustedData(body);
+            if (v == nullptr) {
+                return false;
+            }
+
+            struct AuxDoc : public Doc {
+                AuxDoc(slice body, SharedKeys* sk)
+                : Doc(alloc_slice(body), Doc::kTrusted, sk)
+                {}
+            };
+            AuxDoc doc(body, sk);
+
+            bool ret = false;
+            for (DeepIterator i(doc.asDict()); i; ++i) {
```

Comment:
Can't you use `v->asDict()` instead and avoid having to create the Doc?

Reply:
This is where I don't fully understand it and took the safe path. In this function, I am not sure whether body has already registered with the scope. Has it? If yes, then I don't need to resort to Doc, I think. If not, don't I need to wrap it into a Doc with SharedKeys?

Reply:
Looks like it has not been registered with the scope for the program throws if I use v->asDict(), not wrapping it to a Doc with shared keys.

Reply:
That's unfortunate but I can see why it's happening (for historical reasons.) It's better to use a fleece::impl::Scope, then, because it doesn't require copying the doc body to an alloc_slice.

(Scope is not ref-counted, so you can declare one directly, btw.)